src/app/components/home/home.component.ts

```diff
@@ -75,4 +88,20 @@ export class HomeComponent implements OnInit {
   protected navigateToCreate() {
     this._router.navigate(['create']);
   }
+
+  protected updatePosition() {
+    this.isUpdatingPosition = true
+    this._currentLocation
+      .update()
+      .finally(() => {
+        this.isUpdatingPosition = false;
+      })
+  }
+
+  protected changeCurrentPosition(mapCenter: L.LatLng) {
+    this._currentLocation.setCurrentLocation({
+      latitude: mapCenter.lat,
+      longitude: mapCenter.lng
+    })
+  }
```

Comment:
I would not do this here as it is not exactly view logic. Also the updatingLocation Property could be moved somewhere else